source/MaterialXTest/MaterialXRender/RenderUtil.cpp

```diff
@@ -257,6 +257,11 @@ bool ShaderRenderTester::validate(const mx::FilePathVec& testRootPaths, const mx
                 WARN("Failed to load in file: " + filename.asString() + "See: " + docValidLogFilename + " for details.");
             }
 
+            // For each new file clear the implementation cache.
+            // Since the new file might contain implementations with names
+            // colliding with implementations in previous test cases.
+            context.clearNodeImplementations();
```

Comment:
This fixes a bug in our test suite, where implementation name collisions caused unexpected behavior. Cached implementations where reused unexpectedly.